drivers/bluetooth/hci/spi.c

```diff
@@ -224,26 +231,39 @@ static bool exit_irq_high_loop(void)
 #endif /* CONFIG_BT_SPI_BLUENRG */
 
 #if defined(CONFIG_BT_BLUENRG_ACI)
-static int bt_spi_send_aci_config_data_controller_mode(void)
+static int bt_spi_send_aci_config(uint8_t offset, uint8_t *value, size_t value_len)
 {
-	struct bluenrg_aci_cmd_ll_param *param;
 	struct net_buf *buf;
+	uint8_t *cmd_data;
+	size_t data_len = 2 + value_len;
 
-	buf = bt_hci_cmd_create(BLUENRG_ACI_WRITE_CONFIG_DATA, sizeof(*param));
+	buf = bt_hci_cmd_create(BLUENRG_ACI_WRITE_CONFIG_DATA, data_len);
 	if (!buf) {
 		return -ENOBUFS;
 	}
 
-	param = net_buf_add(buf, sizeof(*param));
-	param->cmd = BLUENRG_ACI_WRITE_CONFIG_CMD_LL;
-	param->length = 0x1;
-	/* Force BlueNRG-MS roles to Link Layer only mode */
-	param->value = BLUENRG_ACI_LL_MODE;
+	cmd_data = net_buf_add(buf, data_len);
+	cmd_data[0] = offset;
+	cmd_data[1] = value_len;
+	memcpy(&cmd_data[2], value, value_len);
 
 	bt_hci_cmd_send(BLUENRG_ACI_WRITE_CONFIG_DATA, buf);
 
 	return 0;
 }
+
+#if defined(CONFIG_BT_HCI_SET_PUBADDR)
+int bt_spi_bluenrg_set_pubaddr(bt_addr_t *addr)
```

Comment:
Since this function is BlueNRG-specific, I guess what you want is to check for CONFIG_BT_BLUENRG_ACI instead of CONFIG_BT_HCI_SET_PUBADDR

Reply:
This is within an `#if defined(CONFIG_BT_BLUENRG_ACI)` guard.

Reply:
> This is within an `#if defined(CONFIG_BT_BLUENRG_ACI)` guard.

Ok, in that case it's redundant, since `BT_BLUENRG_ACI` selects `BT_HCI_SET_PUBADDR`, i.e. it's impossible to disable the option.